docs/guides/mongodb/vault-integration/kmip-encryption/index.md

```diff
@@ -0,0 +1,279 @@
+---
+title: Encryption with Vault KMIP
+menu:
+  docs_{{ .version }}:
+    identifier: guides-mongodb-integration-with-vault-kmip
+    name: Encryption with Vault KMIP
+    parent: guides-mongodb-integration-with-vault
+    weight: 20
+menu_name: docs_{{ .version }}
+section_menu_id: guides
+---
+
+# Encrypt data in KubeDB MongoDB with Hashicorp Vault KMIP Secret Engine.
+
+To demonstrate how to configure KubeDB MongoDB with [HashiCorp Vault KMIP secret engine](https://developer.hashicorp.com/vault/docs/secrets/kmip) for encryption, you can follow this step-by-step example. This documentation will guide you through setting up Vault, configuring the KMIP secret engine, and then configuring KubeDB to use it for MongoDB data encryption.
+
+## Before You Begin
+
+- At first, you need to have a Kubernetes cluster, and the kubectl command-line tool must be configured to communicate with your cluster. If you do not already have a cluster, you can create one by using [kind](https://kind.sigs.k8s.io/docs/user/quick-start/).
+
+- Install KubeDB in your cluster following the steps [here](/docs/setup/README.md).
+
+- HashiCorp Vault instance with the KMIP secret engine enabled.
+
+- To keep things isolated, this tutorial uses a separate namespace called `demo` throughout this tutorial. Run the following command to prepare your cluster for this tutorial:
+
+```bash
+$ kubectl create ns demo
+namespace/demo created
+```
+
+### Setup Hashicorp Vault KMIP secret engine
+
+User can setup Vault KMIP secret engine with [Vault Enterprise](https://developer.hashicorp.com/vault/tutorials/adp/kmip-engine?variants=vault-deploy%3Aenterprise) or [HCP Vault Dedicated](https://developer.hashicorp.com/vault/tutorials/adp/kmip-engine?variants=vault-deploy%3Ahcp).
+For this demo we will use [Hashicorp Cloud Provider(HCP)](https://portal.cloud.hashicorp.com/) Vault Dedicated.
+
+So First we created a `Vault Plus` cluster in HCP. Then we need to configure Vault KMIP according to [this](https://developer.hashicorp.com/vault/tutorials/adp/kmip-engine?variants=vault-deploy%3Ahcp) documentation step by step.
+
+```bash
+$ export VAULT_ADDR=<Public_Cluster_URL>
+$ export VAULT_TOKEN=<Generated_Vault_Token>
+$ export VAULT_NAMESPACE=admin
+
+$ vault secrets enable kmip
+Success! Enabled the kmip secrets engine at: kmip/
+
+$ vault write kmip/config \
+     listen_addrs=0.0.0.0:5696 \
+     server_hostnames=$(echo ${VAULT_ADDR:8} | rev | cut -c6- | rev)
+Success! Data written to: kmip/config
+
+$ vault write -f kmip/scope/finance
+Success! Data written to: kmip/scope/finance
```

Comment:
Add an oneliner comments before each of these commands .. like https://developer.hashicorp.com/vault/tutorials/adp/kmip-engine?variants=vault-deploy%3Ahcp#enable-and-configure-the-kmip-secrets-engine